ra/ra.go

```diff
@@ -988,7 +1006,45 @@ func (ra *RegistrationAuthorityImpl) issueCertificate(
 		return emptyCert, err
 	}
 
-	if ra.publisher != nil {
+	if ra.ctpolicy != nil {
```

Comment:
This conditional body is pretty large, could you pull some of it out into a separate function?